src/frontend/src/components/public/PIAFormTabs/PIAStoredPersonalInformation/index.tsx

```diff
@@ -0,0 +1,843 @@
+    stateChangeHandler(disclosuresOutsideCanada, 'disclosuresOutsideCanada');
+  };
+
+  const listServiceProvidersHeaders = [
+    {
+      name: Messages.AssessmentOfDisclosures.ServiceProviderTableColumnHeaders
+        .Name.en,
+    },
+    {
+      name: Messages.AssessmentOfDisclosures.ServiceProviderTableColumnHeaders
+        .CloudInfrastructure.en,
+    },
+    {
+      name: Messages.AssessmentOfDisclosures.ServiceProviderTableColumnHeaders
+        .StorageDetails.en,
+    },
+  ];
+
+  const [risks, setRisks] = useState<Array<PrivacyRisk>>(
+    storingPersonalInformationForm?.disclosuresOutsideCanada.risks.privacyRisks
+      .length > 0
+      ? storingPersonalInformationForm?.disclosuresOutsideCanada.risks
+          .privacyRisks
+      : [
+          {
+            risk: '',
+            impact: '',
+            likelihoodOfUnauthorizedAccess: '',
+            levelOfPrivacyRisk: '',
+            riskResponse: '',
+            outstandingRisk: '',
+          },
+          {
+            risk: '',
+            impact: '',
+            likelihoodOfUnauthorizedAccess: '',
+            levelOfPrivacyRisk: '',
+            riskResponse: '',
+            outstandingRisk: '',
+          },
+          {
+            risk: '',
+            impact: '',
+            likelihoodOfUnauthorizedAccess: '',
+            levelOfPrivacyRisk: '',
+            riskResponse: '',
+            outstandingRisk: '',
+          },
+          {
+            risk: '',
+            impact: '',
+            likelihoodOfUnauthorizedAccess: '',
+            levelOfPrivacyRisk: '',
+            riskResponse: '',
+            outstandingRisk: '',
+          },
+        ],
+  );
+
+  const [listRisksRows, setListRisksRows] = useState<Array<InputTextProps[]>>(
+    risks.map((risk) => [
+      { value: risk.risk, id: 'one' },
+      { value: risk.impact, id: 'two' },
+      { value: risk.likelihoodOfUnauthorizedAccess, id: 'three' },
+      { value: risk.levelOfPrivacyRisk, id: 'four' },
+      { value: risk.riskResponse, id: 'five' },
+      { value: risk.outstandingRisk, id: 'six' },
+    ]),
+  );
+
+  const addRisksRow = () => {
+    setListRisksRows([
+      ...listRisksRows,
+      [
+        { value: '', id: 'one' },
+        { value: '', id: 'two' },
+        { value: '', id: 'three' },
+        { value: '', id: 'four' },
+        { value: '', id: 'five' },
+        { value: '', id: 'six' },
+      ],
+    ]);
+    setRisks([
+      ...risks,
+      {
+        risk: '',
+        impact: '',
+        likelihoodOfUnauthorizedAccess: '',
+        levelOfPrivacyRisk: '',
+        riskResponse: '',
+        outstandingRisk: '',
+      },
+    ]);
+  };
+
+  const removeRisksRow = (index: number) => {
+    const newData = [...listRisksRows];
+    newData.splice(index, 1);
+    setListRisksRows(newData);
+    risks.splice(index, 1);
+    setRisks(risks);
+    stateChangeHandler(disclosuresOutsideCanada, 'disclosuresOutsideCanada');
```

Comment:
isn't ideal. But I have asked Bing to also move this logic inside List component. Maybe after this tab - we may need this as a cleanup.